src/go/k8s/internal/testutils/k3s.go

```diff
@@ -0,0 +1,605 @@
+}
+
+type K3sStackOptions struct {
+	K3sImage            string
+	CertManagerURL      string
+	CertManagerRepo     string
+	CertManagerChart    string
+	CertManagerVersion  string
+	Driver              string
+	InstallationTimeout time.Duration
+}
+
+func (k K3sStackOptions) normalize() K3sStackOptions {
+	if k.K3sImage == "" {
+		k.K3sImage = defaultK3sImage
+	}
+
+	if k.CertManagerURL == "" {
+		k.CertManagerURL = defaultCertManagerURL
+	}
+
+	if k.CertManagerRepo == "" {
+		k.CertManagerRepo = defaultCertManagerRepo
+	}
+
+	if k.CertManagerChart == "" {
+		k.CertManagerChart = defaultCertManagerChart
+	}
+
+	if k.InstallationTimeout == 0 {
+		k.InstallationTimeout = defaultTimeout
+	}
+
+	return k
+}
+
+type RedpandaOptions struct {
+	Name        string
+	Namespace   string
+	HelmURL     string
+	HelmRepo    string
+	HelmChart   string
+	HelmVersion string
+	Values      map[string]interface{}
+}
+
+func (r RedpandaOptions) normalize(suffix int32) RedpandaOptions {
+	defaultNameAndNamespace := fmt.Sprintf("k3s-%d-%d", time.Now().Unix(), suffix)
+
+	if r.Name == "" {
+		r.Name = defaultNameAndNamespace
+	}
+
+	if r.Namespace == "" {
+		r.Namespace = defaultNameAndNamespace
+	}
+
+	if r.HelmURL == "" {
+		r.HelmURL = defaultHelmURL
+	}
+
+	if r.HelmRepo == "" {
+		r.HelmRepo = defaultHelmRepo
+	}
+
+	if r.HelmChart == "" {
+		r.HelmChart = defaultHelmChart
+	}
+
+	if r.Values == nil {
+		r.Values = map[string]interface{}{}
+	}
+
+	return r
+}
+
+type K3sStack struct {
+	Options       K3sStackOptions
+	config        []byte
+	helm          helmclient.Client
+	directory     string
+	client        client.Client
+	rest          *rest.Config
+	installations atomic.Int32
+	clusters      map[int32]*v1alpha2.Redpanda
+	portForwarder *portForwarder
+	mutex         sync.RWMutex
+
+	ports     []*forwardedPorts
+	portMutex sync.RWMutex
+}
+
+func RunK3sStack(t *testing.T, options K3sStackOptions) *K3sStack {
+	stack := &K3sStack{Options: options.normalize(), clusters: make(map[int32]*v1alpha2.Redpanda)}
+	stack.initializeK3s(t)
+	stack.initializeHelm(t)
+	stack.installCertManagerChart(t)
+
+	stack.portForwarder = newPortForwarder(stack.rest, stack.client)
+	return stack
+}
+
+func (k *K3sStack) InstallCluster(t *testing.T, options RedpandaOptions) *v1alpha2.Redpanda {
+	id := k.installations.Add(1)
+	options = options.normalize(id)
+	k.installRedpandaChart(t, options)
+
+	jsonData, err := json.Marshal(options.Values)
+	require.NoError(t, err)
+
+	cluster := &v1alpha2.Redpanda{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      options.Name,
+			Namespace: options.Namespace,
+		},
+		Spec: v1alpha2.RedpandaSpec{
+			ClusterSpec: &v1alpha2.RedpandaClusterSpec{},
+		},
+	}
+	err = json.Unmarshal(jsonData, &cluster.Spec.ClusterSpec)
+	require.NoError(t, err)
+
+	k.mutex.Lock()
+	defer k.mutex.Unlock()
+
+	k.clusters[id] = cluster
+	return cluster
+}
+
+func (k *K3sStack) Client() client.Client {
+	return k.client
+}
+
+type portForwarder struct {
```

Comment:
Rafal has a port forwarding implementation in the [helm charts](https://github.com/redpanda-data/helm-charts/blob/41cc48195bb690db78739be79906c22c434fd932/pkg/kube/ctl.go#L162). 

We'll need to pick a single repo to start pushing these utilities into so they're more discoverable.